### PR TITLE
Add support for bool type for the numba operator

### PR DESCRIPTION
**Pull request by JanuszL**
- adds support for bool types inside the numba operator
- adds a test for that for CPU and GPU

## Category:
**New feature** (*non-breaking change which adds functionality*)

## Description:
- adds support for bool types inside the numba operator
- adds a test for that for CPU and GPU

### Affected modules and functionalities:
- numba operator

### Key points relevant for the review:
- NA

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
    - new case for test_numba_func .test_numba_func and test_numba_func .test_numba_func_gpu
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Review by banasraf — APPROVED** [review]

File: dali/python/nvidia/dali/plugin/numba/experimental/__init__.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 
 
 _to_numpy = {
+    dali_types.BOOL: "bool_",
```
Comment (mzient) [inline dali/python/nvidia/dali/plugin/numba/experimental/__init__.py:28]: Both `numpy.bool` and `numpy.bool_` are deprecated and removed in newer NumPy. Use plain `bool` instead. This calls for a bit larger rework.

File: dali/test/python/operator_1/test_numba_func.py [hunk 2]
```diff
@@ -42,6 +46,16 @@ def set_all_values_to_255_sample(out0, in0):
...
 
 
+def set_all_values_to_1_sample_gpu(out0, in0):
+    tx, ty, tz = cuda.grid(3)
+    x_s, y_s, z_s = cuda.gridsize(3)
+
+    for z in range(tz, out0.shape[0], z_s):
+        for y in range(ty, out0.shape[1], y_s):
+            for x in range(tx, out0.shape[2], x_s):
+                out0[z][y][x] = 1
```
Comment (mzient) [inline dali/test/python/operator_1/test_numba_func.py:56]: Is it how it's done in numba, or can/should we use more pythonic/numpyic way?
```suggestion
       out0[tz::z_s, ty::y_s, tx::x_s] = 1
```

**Comment by JanuszL** [comment]
!build

**Review by mzient — APPROVED** [review]

**Pull request merged**